### DISCUSSION TITLE
Does the document processing function(rag/svr/task_executor.py) keep running in a loop? Reload the ONNX model every time?

### DISCUSSION THREAD

**Discussion by zzzmc** [infiniflow/ragflow#1927] [Q&A]
Hi, I found the ocr part time-consuming while using it. The model object address printed out new each time, Is a new ONNX model initialized every time a document is parsed? Is it normal to run a "app/naive.py" code loop to parse PDF and require around 3-4GB of memory?
![image](https://github.com/user-attachments/assets/2eb308e3-67d1-4057-97d5-c6982308bb18)

![image](https://github.com/user-attachments/assets/ec817518-db52-4b36-b4d8-83ed9741c9cd)

**Comment by KevinHuSh** [comment]
OCR is time consuming but it only runs if necessary, like, unable to read original text from PDF.
It seems abnormal to use 3-4G by only runing naive.py. We gona check it.

**Reply by zzzmc** [reply]
Thanks for your reply!
What I'm wondering is that "layout_recognize=True" will load a bunch of onnx models. With or without OCR, these ONNX models are going to be reinitialized and loaded, right? No singleton pattern or model uninstall?
Because each time the document is parsed, the ONNX models address printed out is new. I am not sure whether my understanding is correct, so I need your help.

**Reply by zzzmc** [reply]
@KevinHuSh Can you help me with my question?

**Reply by KevinHuSh** [reply]
Even though it re-loaded over times for parsing different files, it would not need that large memory and consume much time since they're all small models less than 10Mb in term of byte size.
